inst/pages/14_alpha_diversity.qmd

```diff
@@ -108,16 +108,84 @@ plotColData(tse,
+
 ```
 
-## Visualizing alpha diversities
+### Diversity {#sec-estimate-diversity}  
+
+The main function, `estimateDiversity`, calculates the selected
+diversity index based on the selected assay data.  
 
+```{r estimate-shannon}
+tse <- mia::estimateDiversity(tse, 
+                              assay.type = "counts",
+                              index = "shannon", 
+                              name = "shannon")
+head(tse$shannon)
+```
 
-### Comparing alpha diversity measures 
+Alpha diversities can be visualized with boxplot. Here, Shannon index is compared 
+between different sample type groups. Individual data points are visualized by 
+plotting them as points with `geom_jitter`.
+
+`geom_signif` is used to test whether these differences are statistically significant.
+It adds p-values to plot.
+
+```{r visualize-shannon}
+library(ggsignif)
+library(ggplot2)
+library(patchwork)
+
+# Subsets the data. Takes only those samples that are from feces, skin, or tongue,
+# and creates data frame from the collected data
+df <- as.data.frame(colData(tse)[tse$SampleType %in% 
+                 c("Feces", "Skin", "Tongue"), ])
+
+# Changes old levels with new levels
+df$SampleType <- factor(df$SampleType)
+
+# For significance testing, all different combinations are determined
+comb <- split(t(combn(levels(df$SampleType), 2)), 
+           seq(nrow(t(combn(levels(df$SampleType), 2)))))
+
+ggplot(df, aes(x = SampleType, y = shannon)) +
+  # Outliers are removed, because otherwise each 
+  # data point would be plotted twice; 
+  # as an outlier of boxplot and as a point of dotplot.
+  geom_boxplot(outlier.shape = NA) + 
+  geom_jitter(width = 0.2) + 
+  geom_signif(comparisons = comb, map_signif_level = FALSE,
+              correction="fdr") + #corrects the p-values
+  theme(text = element_text(size = 10))
+```
+
+### Faith phylogenetic diversity {#sec-faith-diversity}
+
+The Faith index is returned by the function `estimateFaith`.
+
+```{r phylo-div-1}
+tse <- mia::estimateFaith(tse,
+                          assay.type = "counts")
+head(tse$faith)
+```
+
+**Note**: because `tse` is a `TreeSummarizedExperiment` object, its phylogenetic tree is used by default. However, the optional argument `tree` must be provided if `tse` does not contain one.
+
+Below a visual comparison between shannon and faith indices is shown with a violin plot.
+
+```{r phylo-div-2}
+plots <- lapply(c("shannon", "faith"),
+                plotColData,
+                object = tse, colour_by = "SampleType")
+plots[[1]] + plots[[2]] +
+  plot_layout(guides = "collect")
+```
+ 
+Alternatively, the phylogenetic diversity can be calculated by `mia::estimateDiversity`. This is a faster re-implementation of   
+the widely used function in _`picante`_ [@R_picante, @Kembel2010].  
 
-Let us now compare the estimated Shannon and Faith indices across
-samples. First we add Shannon and Faith indices.
+Load `picante` R package and get the `phylo` stored in `rowTree`.
```

Comment:
What is the intention of this? It seems that picante package is not loaded even though it says like that

Reply:
I don't know. It's not something I have changed. That's a previous commit.

Reply:
For some reason it shows as part of this commit.

Have you synced your branch with the latest version? 

Anyway I thing that line could be removed.

Reply:
I now figure out how to sync branches. It says "up to date".

Reply:
Do you use normal git merge when updating with devel branch?

Reply:
Used to. Now I usually use `git rebase`.

Reply:
I will now just create new branches from devel and merge those directly on this repo. I think it's be easier.